Source/WebCore/rendering/mathml/RenderMathMLRoot.cpp

```diff
@@ -313,6 +312,14 @@ void RenderMathMLRoot::paint(PaintInfo& info, const LayoutPoint& paintOffset)
     info.context().drawLine(ruleOffsetFrom, ruleOffsetTo);
 }
 
+std::optional<LayoutUnit> RenderMathMLRoot::firstLineBaseline() const
+{
+    if (!isValid())
+        return RenderMathMLRow::firstLineBaseline();
+
+    return RenderMathMLBlock::firstLineBaseline();
```

Comment:
So this what is causing a behavior change, at least visible for `radical-fallback.html`.

Before this patch `RenderMathMLRoot::firstLineBaseline()` did not exist but `RenderMathMLRoot` was already inheriting from `RenderMathMLRow` (contrary to fraction/scripts). Now, invalid mroot markup continues to use `RenderMathMLRow::firstLineBaseline()` but msqrt (always valid) and mroot will use `RenderMathMLBlock::firstLineBaseline()` instead.

Checking the two methods, I believe keep `RenderMathMLRow::firstLineBaseline()` (using the baseline of the first child as the baseline of the whole msqrt/mroot) is also correct (see links below). So we don't need to override `firstLineBaseline()` here.

> https://w3c.github.io/mathml-core/#figure-box-msqrt
> The [alphabetic baseline](https://w3c.github.io/mathml-core/#dfn-alphabetic-baseline) of the [msqrt base](https://w3c.github.io/mathml-core/#dfn-msqrt-base) is aligned with the [alphabetic baseline](https://w3c.github.io/mathml-core/#dfn-alphabetic-baseline).

> https://w3c.github.io/mathml-core/#figure-box-mroot
> The [alphabetic baseline](https://w3c.github.io/mathml-core/#dfn-alphabetic-baseline) of B is aligned with the [alphabetic baseline](https://w3c.github.io/mathml-core/#dfn-alphabetic-baseline).

> https://w3c.github.io/mathml-core/#layout-of-mrow
> For each [in-flow](https://www.w3.org/TR/css-display-3/#in-flow) child ... its [block offset](https://w3c.github.io/mathml-core/#dfn-block-offset) such that the [alphabetic baseline](https://w3c.github.io/mathml-core/#dfn-alphabetic-baseline) of the child is aligned with the [alphabetic baseline](https://w3c.github.io/mathml-core/#dfn-alphabetic-baseline).

Reply:
I see, thank you!